filepicker/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerViewModel.kt

```diff
@@ -190,6 +191,17 @@ internal class ArkFilePickerViewModel(
         val roots = rootsWithFavorites.keys
         val root = roots.find { root -> file.startsWith(root) }
         val favorites = rootsWithFavorites[root]?.flatten()
+
+        val hasNestedRoot = roots.contains(file)
+                || (roots.indexOfFirst { path ->
+                    val index = path.toString().indexOf(file.toString())
+                    (index >= 0) && (path.toString()[index] == '/') } >= 0)
```

Comment:
Can you make second condition as separate variable?
Second condition is hard to understand, maybe `root.startsWith(file) || file.startsWith(root)`
`startsWith` checks if this is child of argument path

Reply:
It seems you don't check if selected folder is parent of root.
We can't allow `/Internal/Music` root and `/Internal/Music/A` selected folder to pass

Reply:
@mdrlzy The changes in this PR are doing exactly what you think is missing.
That means, if `/Internal/Music/A` is root, pinning (in consequence, making root) `/Internal/Music` is disallowed.